internal/testutil/testutil.go

```diff
@@ -0,0 +1,102 @@
+package testutil
+
+import (
+	"net"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"go.sia.tech/core/consensus"
+	"go.sia.tech/core/gateway"
+	"go.sia.tech/core/types"
+	"go.sia.tech/coreutils/chain"
+	"go.sia.tech/coreutils/syncer"
+	"go.sia.tech/coreutils/testutil"
+	"go.sia.tech/walletd/persist/sqlite"
+	"go.uber.org/zap"
+)
+
+type (
+	// A ConsensusNode is a test harness for starting a bare-bones consensus node.
+	ConsensusNode struct {
+		Store  *sqlite.Store
+		Chain  *chain.Manager
+		Syncer *syncer.Syncer
+	}
+)
+
+// WaitForSync waits for the store to sync to the current tip of the chain manager.
+func (cn *ConsensusNode) WaitForSync(tb testing.TB) {
+	tb.Helper()
+
+	for i := 0; i < 1000; i++ {
+		index, err := cn.Store.LastCommittedIndex()
+		if err != nil {
+			tb.Fatal(err)
+		} else if index == cn.Chain.Tip() {
+			return
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+	tb.Fatal("timeout waiting for sync")
+}
+
+// MineBlocks mines n blocks, sending the rewards to addr.
+func (cn *ConsensusNode) MineBlocks(tb testing.TB, addr types.Address, n int) {
+	tb.Helper()
+
+	for i := 0; i < n; i++ {
+		testutil.MineBlocks(tb, cn.Chain, addr, 1)
+		cn.WaitForSync(tb)
+	}
+}
+
+// NewConsensusNode creates a new ConsensusNode.
+func NewConsensusNode(tb testing.TB, n *consensus.Network, genesis types.Block, log *zap.Logger) *ConsensusNode {
```

Comment:
this is great but could've been an entirely separate PR (maybe I missed it though by being OOO and this got reviewed separately)

Reply:
> The test diff does not spark joy, but I don't want to split it out. The only one with material changes is the new one at the bottom